code/game/objects/items/devices/nyamagotchi.dm

```diff
@@ -0,0 +1,184 @@
+/obj/item/nyamagotchi
+	name = "nyamagotchi"
+	icon = 'icons/obj/devices/nyamagotchi.dmi'
+	desc = "A small electronic 'pet' that requires care and attention. An ancient relic sure to evoke nostalgic feelings."
+	icon_state = "default"
+	worn_icon_state = "nothing"
+	base_icon_state = "default"
+	slot_flags = ITEM_SLOT_BELT
+	w_class = WEIGHT_CLASS_SMALL
+	force = 2
+	throwforce = 2
+	lefthand_file = 'icons/mob/inhands/items/devices_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/items/devices_righthand.dmi'
+	inhand_icon_state = "electronic"
+	custom_materials = list(/datum/material/iron=SMALL_MATERIAL_AMOUNT, /datum/material/glass=SMALL_MATERIAL_AMOUNT, /datum/material/plastic=SMALL_MATERIAL_AMOUNT)
+	// interaction_flags_atom = parent_type::interaction_flags_atom | INTERACT_ATOM_ALLOW_USER_LOCATION | INTERACT_ATOM_IGNORE_MOBILITY
+	var/list/icons_available = list()
+	var/radial_icon_file = 'icons/hud/radial_nyamagotchi.dmi'
+
+	/// Hunger level (0 = full, 100 = starving)
+	var/hunger = 0
+	/// Happiness level (0 = sad, 100 = very happy)
+	var/happiness = 100
+	var/energy = 0       // Energy level (0 = tired, 100 = full of energy)
+	var/age = 0            // Age in "days" or some unit of time
+	var/alive = 0          // Status flag (1 = alive, 0 = dead)
```

Comment:
Unless I missed it reading your code, you are always using these alive/died vars together.

Instead, make it a single variable, and throw in a local define: (you don't need to define DEAD, because that one is already done upstream on TG)
```
#define ALIVE 1
```
at the top of the dm, and
```
#undef ALIVE
```
at the end.

Reply:
I added a 'died' variable for inspecting it for the first time before startup, but it's not that big of a deal.